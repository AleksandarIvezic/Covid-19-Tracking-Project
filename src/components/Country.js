import React from 'react';
import { useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';
import '../styling/Country.css';
import countryIso from '../libraries/countryIso';

const Country = ({ country, totalConfirmed, handleClick }) => {
  const history = useHistory();
  const handleRoute = (route) => {
    history.push(route);
  };

  const iso = countryIso[country];

  const link = iso ? `https://raw.githubusercontent.com/djaiss/mapsicon/33ba28808f8d32b5bae0ffada9cadd07073852e1/all/${iso.toLowerCase()}/vector.svg` : '/World_map_svg.svg';
  const nameCountry = country.length > 18 ? country.split(' ')[0] : country;

  return (
    <button
      className="country position-relative country text-light"
      type="button"
      onClick={() => {
        handleRoute('/country');
        handleClick(country);
      }}
    >
      <img className="c-img" src={link} alt="country" width="100" height="100" />
      <div className="bottom d-flex justify-content-end">
        <div className="text d-flex flex-column justify-content-center">
          <span className="fw-bold text-uppercase ">{nameCountry}</span>
          <span>{Intl.NumberFormat('de-DE').format(totalConfirmed)}</span>
        </div>
      </div>
      <i className="far fa-arrow-alt-circle-right position-absolute top-0 end-0 p-2"> </i>
    </button>
  );
};

Country.propTypes = {
  country: PropTypes.string.isRequired,
  totalConfirmed: PropTypes.number.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default Country;
