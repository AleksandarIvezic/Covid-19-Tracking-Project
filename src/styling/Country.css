.country {
  height: 150px;
  width: 100%;
}

.c-img {
  filter: invert(100%) sepia(0%) saturate(7483%) hue-rotate(353deg) brightness(97%) contrast(107%);
}

button {
  background-color: transparent;
  border: none;
}
