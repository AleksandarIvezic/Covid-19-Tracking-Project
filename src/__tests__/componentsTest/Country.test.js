import React from 'react';
import renderer from 'react-test-renderer';
import Country from '../../components/Country';

describe('Country', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Country country="Serbia" totalConfirmed={7} handleClick={() => 2 + 3} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
