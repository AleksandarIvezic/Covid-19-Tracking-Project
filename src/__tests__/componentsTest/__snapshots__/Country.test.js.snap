// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country renders correctly 1`] = `
<button
  className="country position-relative country text-light"
  onClick={[Function]}
  type="button"
>
  <img
    alt="country"
    className="c-img"
    height="100"
    src="https://raw.githubusercontent.com/djaiss/mapsicon/33ba28808f8d32b5bae0ffada9cadd07073852e1/all/rs/vector.svg"
    width="100"
  />
  <div
    className="bottom d-flex justify-content-end"
  >
    <div
      className="text d-flex flex-column justify-content-center"
    >
      <span
        className="fw-bold text-uppercase "
      >
        Serbia
      </span>
      <span>
        7
      </span>
    </div>
  </div>
  <i
    className="far fa-arrow-alt-circle-right position-absolute top-0 end-0 p-2"
  >
     
  </i>
</button>
`;
